app/mixins/routes/paginated.js

```diff
@@ -0,0 +1,39 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create({
+  setPaginationMeta: function(controller, type){
+    var meta = this.store.metadataFor(type);
+
+    controller.set('currentPage', meta.current_page);
+    controller.set('totalCount', meta.total_count);
+    controller.set('perPage', meta.per_page);
+  },
+
+  actions: {
+    goToPage: function(page){
+      var controller = this.controller;
+      var store = this.store;
+      var route = this;
+      var query = { page: page };
+
+      var paginatedResourceOwner = this.controller.get('paginatedResourceOwner');
+      query[ paginatedResourceOwner.constructor.typeKey ] = paginatedResourceOwner;
+
+      store.find('operation', query).then(function(operations){
```

Comment:
I might expect that the query to load a new page is a hook called on the route. So a route must include the mixin, and define a hook for fetching new pages. That way all the `store` related work stays on the route and this mixin is only responsible for managing convenience actions (next/prev page) and passing params to hooks on the route.